proxy/director.go

```diff
@@ -126,7 +126,7 @@ func timedUnavailabilityFunc(wait time.Duration) func(*endpoint) {
 	return func(ep *endpoint) {
 		time.AfterFunc(wait, func() {
 			ep.Available = true
-			log.Printf("proxy: marked endpoint %s available", ep.URL.String())
+			log.Printf("proxy: marked endpoint %s available to retry it until refresh interval time arrived.", ep.URL.String())
```

Comment:
I misunderstood the meaning of refresh interval, and should use failure-wait time instead. Let rephrase what I said.

proxy keeps some cached endpoints, and set availability on them to avoid meaningless retry. At the meantime, it refreshes the endpoints every refresh interval.

So the endpoint attempt workflow is
1. proxy fails to get response from the endpoint
2. proxy marks endpoint as unavailable
3. proxy will not try the endpoint in failure-wait time
4. proxy marks endpoint as available to hope it back online

The refresh interval is a separate thing. It is used to refresh cached endpoints by fetching the advertised client urls from etcd cluster. The proxy in example may cache endpoint for refresh interval(30s). When it finds that it cannot refresh from etcd cluster, it thinks no endpoint exists at this point, and clears the cache that may be available to redirect requests.